fix: updated auth with socket.io explanation (#43066)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/authentication-with-socket.io.md

);
```

> You could note that configuring passport authentication for *Socket.IO* is very similar to configuring `session` middleware for our API. This is true because they are meant to use same authentication method - get session id from cookie and validate it.

Reviewer: ```suggestion
Note that configuring Passport authentication for Socket.IO is very similar to the way we configured the `session` middleware for the API. This is because they are meant to use the same authentication method — get the session id from a cookie and validate it.
```

> Beforehand for configuring `session` middleware we didn't set cookie name for session (`key`) explicitly - `express-session` package was using default value.

Reviewer: ```suggestion
Before when we configured the `session` middleware we didn't set the cookie name for session (`key`) explicitly. This is because the `session` package was using the default value.
```

> But now as we have added another package which needs access to same value from cookies we need to set `key` value explicitly in both configuration objects. So be sure to add the `key` with the cookie name to the `session` middleware to match the *Socket.IO* key as well as the `store` reference to the options, near where we set `saveUninitialized: true`. This is necessary to tell *Socket.IO* which session to relate to.

Reviewer: I tried to break this up a little bit, and to clarify the second paragraph a bit:

```suggestion
But now that we've added another package which needs access to the same value from the cookies, we need to set the `key` value explicitly in both configuration objects.

Be sure to add the `key` with the cookie name to the `session` middleware that matches the Socket.IO key. Also, add the `store` reference to the options, near where we set `saveUninitialized: true`. This is necessary to tell Socket.IO which session to relate to.
```

);
```

Note that configuring Passport authentication for Socket.IO is very similar to the way we configured the `session` middleware for the API. This is because they are meant to use the same authentication method — get the session id from a cookie and validate it.

> Before when we configured the `session` middleware we didn't set the cookie name for session (`key`) explicitly. This is because the `session` package was using the default value. But now that we've added another package which needs access to the same value from the cookies, we need to set the `key` value explicitly in both configuration objects.

Reviewer 2: ```suggestion
Previously, when we configured the `session` middleware, we didn't explicitly set the cookie name for session (`key`). This is because the `session` package was using the default value. Now that we've added another package which needs access to the same value from the cookies, we need to explicitly set the `key` value in both configuration objects.
```
